webui.py

```diff
@@ -189,51 +198,80 @@ def update_dependencies():
-        if gptq_min_compute_check or compute_array.returncode != 0:
+        if '+rocm' in torver:
+            run_cmd("git clone https://github.com/WapaMario63/GPTQ-for-LLaMa-ROCm.git GPTQ-for-LLaMa -b rocm", assert_success=True, environment=True)
+        elif gptq_min_compute_check or compute_array.returncode != 0:
             run_cmd("git clone https://github.com/oobabooga/GPTQ-for-LLaMa.git -b cuda", assert_success=True, environment=True)
         else:
             run_cmd("git clone https://github.com/qwopqwop200/GPTQ-for-LLaMa.git -b cuda", assert_success=True, environment=True)
 
-    # Install GPTQ-for-LLaMa dependencies
-    os.chdir("GPTQ-for-LLaMa")
-    run_cmd("git pull", assert_success=True, environment=True)
-
     # On some Linux distributions, g++ may not exist or be the wrong version to compile GPTQ-for-LLaMa
     if sys.platform.startswith("linux"):
         gxx_output = run_cmd("g++ -dumpfullversion -dumpversion", environment=True, capture_output=True)
         if gxx_output.returncode != 0 or int(gxx_output.stdout.strip().split(b".")[0]) > 11:
             # Install the correct version of g++
             run_cmd("conda install -y -k gxx_linux-64=11.2.0", environment=True)
 
+    # Install/Update ROCm AutoGPTQ for AMD GPUs
+    if '+rocm' in torver:
+        if run_cmd("[ -d ./AutoGPTQ-rocm ] && rm -rfd ./AutoGPTQ-rocm; git clone https://github.com/jllllll/AutoGPTQ.git ./AutoGPTQ-rocm -b rocm && cp ./AutoGPTQ-rocm/setup_rocm.py ./AutoGPTQ-rocm/setup.py && python -m pip install ./AutoGPTQ-rocm --force-reinstall --no-deps", environment=True).returncode != 0:
+            print_big_message("WARNING: AutoGPTQ kernel compilation failed!\n       The installer will proceed to install a pre-compiled wheel.")
+            if run_cmd("python -m pip install https://github.com/jllllll/GPTQ-for-LLaMa-Wheels/raw/Linux-x64/ROCm-5.4.2/auto_gptq-0.3.2%2Brocm5.4.2-cp310-cp310-linux_x86_64.whl --force-reinstall --no-deps", environment=True).returncode != 0:
+                print_big_message("ERROR: AutoGPTQ wheel installation failed!\n       You will not be able to use GPTQ-based models with AutoGPTQ.")
+
+    # Install GPTQ-for-LLaMa dependencies
+    os.chdir("GPTQ-for-LLaMa")
+    run_cmd("git pull", environment=True)
+    
+    # Finds the path to your dependencies
+    for sitedir in site.getsitepackages():
+        if "site-packages" in sitedir:
+            site_packages_path = sitedir
+            break
+
+    # This path is critical to installing the following dependencies
+    if site_packages_path is None:
+        print("Could not find the path to your Python packages. Exiting...")
+        sys.exit()
+
     # Compile and install GPTQ-for-LLaMa
-    if os.path.exists('setup_cuda.py'):
+    if '+rocm' in torver:
+        if os.path.exists('setup_rocm.py'):
+            os.replace("setup_rocm.py", "setup.py")
+        # Skip compile for AMD GPU if wheel is successfully installed
+        if rocm_gptq:
```

Comment:
Is this variable (`rocm_gptq`) undefined? `pyflakes` complained about it

Reply:
Could have sworn I removed that. Guess I forgot to.

It is a holdover from when I was initially defaulting to installing wheels. I switched to trying to build locally first as this is the behavior of the CUDA side of the installer and will potentially make AMD support more reliable.

Reply:
Thanks for the PR -- I figured it was something left over. I have already accepted it and updated the installers :)